### PR TITLE
Create Doxygen documentation for orog_mask_tools.fd/lake.fd/enclosure_cnvx

### LINKED ISSUE #313: complete documentation of sorc/orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90
Work with @GeorgeGayno-NOAA to fill in information.

### PR DESCRIPTION
**Pull request by kgerheiser**
Fixes #313 

@GeorgeGayno-NOAA what is `co_gc`?

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Please also include the name of the utility in PR titles.

File: sorc/orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90 [hunk 1]
```diff
@@ -32,15 +32,17 @@ PROGRAM testenc
     END PROGRAM
 #endif
 
-
+!> Test whether a given point, p, is inside a convex spherical polygon defined with a series of n vertices
```
Comment (edwardhartnett) [inline sorc/orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90:35]: Add @author and @brief tags at the file leve.

I see there is some test code here. In such cases create an issue titled something like: "recover test code in orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90 as unit test"

File: sorc/orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90 [hunk 2]
```diff
@@ -32,15 +32,17 @@ PROGRAM testenc
...
+!! Both the test point and the polygon are specified in latitude and longitude radians
```
Comment (edwardhartnett) [inline sorc/orog_mask_tools.fd/lake.fd/enclosure_cnvx.F90:36]: Punctuation at the end of all sentences please. (Why is this missing from so much of our fortran documentation? Did we used to charge extra for programmers to use the '.' key?)

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA can you answer Kyle's question so we can get this merged?

**Comment by GeorgeGayno-NOAA** [comment]
> Fixes #313
> 
> @GeorgeGayno-NOAA what is `co_gc`?

Let me take a look.

**Comment by GeorgeGayno-NOAA** [comment]
@shansun6 Can you define the variable co_gc for us?

**Comment by GeorgeGayno-NOAA** [comment]
Here is the definition of 'co_gc' from Ning Wang:

```
The following is the definition of co_gc:

co_gc, integer
  = i: if test point is on or within the distance of epsilon (0.000000000000001D0 currently) to side i of the given 
spherical polygon ;
  = 0: if test point is not on any side of the given spherical polygon.

*It is assumed that a side of a spherical polygon is a great-circle arc that connects 2 adjacent vertices of the polygon.

The output variable co_gc from enclosure_cnvx( ) is not being used by lakefrac program, which only needs to know 
if a test point is inside a given spherical quadrilateral.
```

**Comment by edwardhartnett** [comment]
Kyle if you can insert George's answer into the documentation somewhere, we can get this merged.

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**